R/mobr_boxplots.R

```diff
@@ -154,61 +169,20 @@ These are removed for the calculation of rarefied richness."))
       F_rand$betaPIE[i] = anova(lm(betaPIE_sample ~ group_id_rand))$F[1]
    }
    
-   p_S       = sum(F_obs$S       < F_rand$S)/nperm
-   p_N       = sum(F_obs$N       < F_rand$N)/nperm
-   p_S_rare  = sum(F_obs$S_rare  < F_rand$S_rare)/nperm
-   p_PIE     = sum(F_obs$PIE     < F_rand$PIE)/nperm
-   p_ENS_PIE = sum(F_obs$ENS_PIE < F_rand$ENS_PIE)/nperm
-   p_S_asymp = sum(F_obs$S_asymp < F_rand$S_asymp)/nperm
-   p_betaPIE = sum(F_obs$betaPIE < F_rand$betaPIE)/nperm
-   
-   if (plot == T){
-      #windows(10,6)
-      op = par(mfcol = c(3,5), las = 1, font.main = 1)
-      
-      # PIE
-      par(fig = c(0.2, 0.4, 0.66, 1))
-      panel_title = paste("PIE (p = ", p_PIE,")", sep = "")
-      boxplot(PIE_sample ~ group_id, main = panel_title)
-      
-      par(fig = c(0.4, 0.6, 0.66, 1), new = T)
-      panel_title = paste("beta PIE (p = ", p_betaPIE,")", sep = "")
-      boxplot(betaPIE_sample ~ group_id, main = panel_title)
-      
-      par(fig = c(0.6, 0.8, 0.66, 1), new = T)
-      boxplot(PIE_group ~ levels(group_id), main = "PIE", boxwex = 0)
-      points(PIE_group, pch =19)
-      
-      # S observed samples
-      par(fig = c(0, 0.2, 0.33, 0.66), new = T)
-      panel_title = paste("S (p = ", p_S,")", sep = "")
-      boxplot(S_sample ~ group_id, main = panel_title)
-      
-      # N samples
-      par(fig = c(0.2, 0.4, 0.33, 0.66), new = T)
-      panel_title = paste("N (p = ", p_N,")", sep = "")
-      boxplot(N_sample ~ group_id, main = panel_title)
-      
-      # N groups
-      par(fig = c(0.6, 0.8, 0.33, 0.66), new = T)
-      boxplot(N_group ~ levels(group_id), main = "N", boxwex = 0)
-      points(N_group, pch = 19)
-      
-      # S groups
-      par(fig = c(0.8, 1.0, 0.33, 0.66), new = T)
-      boxplot(S_group ~ levels(group_id), main = "S obs", boxwex = 0)
-      points(S_group, pch = 19)
-      
-      # S asymptotic
-      par(fig = c(0.2, 0.4, 0, 0.33), new = T)
-      panel_title = paste("S asymp (p = ", p_S_asymp,")", sep = "")
-      boxplot(S_asymp_sample ~ group_id, main = panel_title)
-      
-      par(fig = c(0.6, 0.8, 0, 0.33), new = T)
-      boxplot(S_asymp_group ~ levels(group_id), main = "asymptotic S", boxwex = 0)
-      points(S_asymp_group, pch =19)
-   }
-   
+   p_S       = sum(F_obs$S       <= F_rand$S) / nperm
+   p_N       = sum(F_obs$N       <= F_rand$N) / nperm
+   p_S_rare  = sum(F_obs$S_rare  <= F_rand$S_rare) / nperm
+   p_PIE     = sum(F_obs$PIE     <= F_rand$PIE) / nperm
+   p_ENS_PIE = sum(F_obs$ENS_PIE <= F_rand$ENS_PIE) / nperm
+   p_S_asymp = sum(F_obs$S_asymp <= F_rand$S_asymp) / nperm
+   p_betaPIE = sum(F_obs$betaPIE <= F_rand$betaPIE) / nperm
```

Comment:
Following the above comment, I think `F_obs$stat` should be compared with only the permutations, excluding itself. So I'm thinking it should read ```sum(F_obs$stat <= F_rand$stat[2:(nperm+1)]) / nperm```.